### PR TITLE
Appify testify

### LINKED ISSUE #3: default current_app.conf.REDBEAT_REDIS_URL if not set

### PR DESCRIPTION
**Pull request by sibson**
fixes #3, #6.  Mostly involves removing the usage of current_app and thus allowing us to better follow proper celery design and help ease testing.

Might be easier to follow individual commits than the full PR.  Slight changes to the API but I think its cleaner overall.

[Review on Reviewable](https://reviewable.io/reviews/sibson/redbeat/16)

### CODE REVIEW

**Comment by sibson** [comment]
fixes #3, #6

**Comment by lambacck** [comment]
Reviewed 3 of 6 files at r1, 4 of 5 files at r2.
Review status: 7 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[redbeat/schedulers.py, line 103 \[r2\]](https://reviewable.io:443/reviews/sibson/redbeat/16#-K5HagpDqTC_Zgkx82L0:-K5HagpDqTC_Zgkx82L1:1253777573) ([raw file](https://github.com/sibson/redbeat/blob/dfcaa1d68aa83e4a78e10152e836fb623cf4e53f/redbeat/schedulers.py#L103)):*
This seems like it won't change for the life of the class instance. Maybe we should use something like [reify from pyramid](https://github.com/Pylons/pyramid/blob/7d5d899f5043b381348bd349792fda6da957ab50/pyramid/decorator.py) instead of property which memoizes the redis client instance for the duration of this object instance. Might even want even want to think about caching it on the app.

---

*[redbeat/schedulers.py, line 116 \[r2\]](https://reviewable.io:443/reviews/sibson/redbeat/16#-K5Hc39XRWuYleVUdA2_:-K5Hc39XRWuYleVUdA2a:-958316384) ([raw file](https://github.com/sibson/redbeat/blob/dfcaa1d68aa83e4a78e10152e836fb623cf4e53f/redbeat/schedulers.py#L116)):*
The reify comment is driven by things like this where multiple calls to the redis method are masked as attribute lookup.

---

*[redbeat/schedulers.py, line 167 \[r2\]](https://reviewable.io:443/reviews/sibson/redbeat/16#-K5HcRy-BqoXiYMzFUqA:-K5HcRy-BqoXiYMzFUqB:1034887063) ([raw file](https://github.com/sibson/redbeat/blob/dfcaa1d68aa83e4a78e10152e836fb623cf4e53f/redbeat/schedulers.py#L167)):*
same here for reify.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sibson/redbeat/16)*

**Comment by sibson** [comment]
Review status: 7 of 8 files reviewed at latest revision, 3 unresolved discussions.

---

*[redbeat/schedulers.py, line 103 \[r2\]](https://reviewable.io:443/reviews/sibson/redbeat/16#-K5HagpDqTC_Zgkx82L0:-K5ISXEz1qaqH9wlrqEB:-1943454210) ([raw file](https://github.com/sibson/redbeat/blob/dfcaa1d68aa83e4a78e10152e836fb623cf4e53f/redbeat/schedulers.py#L103)):*
I'm not happy as it stands either.  I'd thought about caching on the app, but wasn't sure if that was "good" celery design.  This seemed like the simplest solution for this PR with the plan to revist once the dust settles.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sibson/redbeat/16)*

**Comment by sibson** [comment]
Review status: 4 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[redbeat/schedulers.py, line 103 \[r2\]](https://reviewable.io:443/reviews/sibson/redbeat/16#-K5HagpDqTC_Zgkx82L0:-K8bZxHagtWuD735K4ke:1706342915) ([raw file](https://github.com/sibson/redbeat/blob/dfcaa1d68aa83e4a78e10152e836fb623cf4e53f/redbeat/schedulers.py#L103)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sibson/redbeat/16)*

File: README.rst [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-  1. Dynamic task creation and modification, no restart required
-  2. Externally manage tasks from any language
-  3. Shared data store; Beat isn't tied to a single drive
-  4. Fast startup
-  5. Avoid running multiple Beat servers
+  1. Dynamic live task creation and modification, without lengthy downtime
+  2. Externally manage tasks from any language with Redis bindings
+  3. Shared data store; Beat isn't tied to a single drive or machine
+  4. Fast startup even with a large task count
+  5. Prevent accidental running multiple Beat servers
```
Comment (lambacck) [inline README.rst]: accidentally

File: redbeat/schedulers.py [hunk 2]
```diff
@@ -93,41 +112,36 @@ def save_definition(self):
...
 
-    def save_meta(self):
-        meta = {
-            'last_run_at': self.last_run_at,
-            'total_run_count': self.total_run_count,
-        }
-        rdb.hset(self.key, 'meta', json.dumps(meta, cls=RedBeatJSONEncoder))
+    def delete(self):
+        self.redis.zrem(self.app.conf.REDBEAT_SCHEDULE_KEY, self.key)
+        self.redis.delete(self.key)
```
Comment (lambacck) [inline redbeat/schedulers.py]: delete meta?

Comment (lambacck) [inline redbeat/schedulers.py]: oh, nevermind meta is a key on the key.

**Comment by lambacck** [comment]
Reviewed 4 of 5 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sibson/redbeat/16)*

**Comment by lambacck** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/sibson/redbeat/16)*

**Comment by lambacck** [comment]
:+1:

**Pull request merged**